mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -0,0 +1,93 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.delete;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.in;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.james.backends.cassandra.utils.CassandraUtils;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.table.CassandraAnnotationTable;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+
+public class CassandraAnnotationMapper {
+
+    private final CassandraId mailboxId;
+    private final Session session;
+
+    public CassandraAnnotationMapper(CassandraId mailboxId, Session session) {
+        this.mailboxId = mailboxId;
+        this.session = session;
+    }
+
+    public List<MailboxAnnotation> getAnnotations() {
+        return getResult(getStoredAnnotationRow());
+    }
+
+    public List<MailboxAnnotation> getAnnotations(List<String> keys) {
+        return getResult(getStoredAnnotationRow(keys));
```

Comment:
`getResult()` looks weird : what does it do, getStoredAnnotationRow already return a ResultSet ?
We could introduce an execution method into CassandraUtils that expects to return a Stream of Row.
getStoredAnnotationRow would use this method and return a Stream<Row>.
Finally, getAnnotations would just have have to map the stream to Annotation and you could drop getResult method